File: sts.go
package acos

import (
	"context"

	"github.com/aws/aws-sdk-go-v2/service/iam"
	"github.com/aws/aws-sdk-go-v2/service/sts"
)

var (
	// AWS clients
	stsClient *sts.Client
	iamClient *iam.Client
)

func init() {
	stsClient = sts.NewFromConfig(cfg)
	iamClient = iam.NewFromConfig(cfg)
}

// GetCallerAccount returns the account ID and name for the current user session.
func GetCallerAccount(ctx context.Context) ([]string, error) {
	res := make([]string, 2)
	out, err := stsClient.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
	if err != nil {
		return res, err
	}
	res[0] = *out.Account // Account ID
	// Try to fetch human-readable account name
	out2, err := iamClient.ListAccountAliases(ctx, &iam.ListAccountAliasesInput{})
	if err != nil {
		return res, err
	}
	if len(out2.AccountAliases) > 0 {
		res[1] = out2.AccountAliases[0] // Alias as account name
	} else {
		res[1] = "Name not configured"
	}
	return res, nil
}

File: src/sts.rs
use anyhow::Result;
use aws_sdk_iam::Client as IamClient;
use aws_sdk_sts::Client as StsClient;

/// Get the caller's account ID and name (alias).
/// Returns (account_id, account_name).
pub async fn get_caller_account(
    sts_client: &StsClient,
    iam_client: &IamClient,
) -> Result<(String, String)> {
    let identity = sts_client.get_caller_identity().send().await?;
    let account_id = identity.account().unwrap_or_default().to_string();

    // Try to fetch human-readable account name via IAM alias
    let aliases_output = iam_client.list_account_aliases().send().await?;
    let account_name = if let Some(first) = aliases_output.account_aliases().first() {
        first.clone()
    } else {
        "Name not configured".to_string()
    };

    Ok((account_id, account_name))
}
